File: server/src/models/Order.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/database.js';

const Order = sequelize.define('Order', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  order_number: {
    type: DataTypes.STRING,
    unique: true,
    allowNull: false
  },
  user_id: {
    type: DataTypes.UUID,
    allowNull: true // может быть гостевой заказ
  },
  customer_name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  customer_phone: {
    type: DataTypes.STRING,
    allowNull: false
  },
  customer_email: {
    type: DataTypes.STRING,
    allowNull: true
  },
  order_type: {
    type: DataTypes.ENUM('delivery', 'pickup', 'dine_in'),
    allowNull: false
  },
  delivery_address: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  delivery_coordinates: {
    type: DataTypes.JSONB, // {lat, lng}
    allowNull: true
  },
  items: {
    type: DataTypes.JSONB, // [{item_id, name, quantity, price, options}]
    allowNull: false
  },
  subtotal: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false
  },
  delivery_fee: {
    type: DataTypes.DECIMAL(10, 2),
    defaultValue: 0
  },
  total: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM(
      'pending',      // ожидает подтверждения
      'confirmed',    // подтвержден
      'preparing',    // готовится
      'ready',        // готов
      'delivering',   // доставляется
      'completed',    // завершен
      'cancelled'     // отменен
    ),
    defaultValue: 'pending'
  },
  payment_status: {
    type: DataTypes.ENUM('pending', 'paid', 'failed', 'refunded'),
    defaultValue: 'pending'
  },
  payment_method: {
    type: DataTypes.ENUM('card', 'cash', 'sbp'),
    allowNull: true
  },
  special_instructions: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  scheduled_time: {
    type: DataTypes.DATE,
    allowNull: true
  }
}, {
  tableName: 'orders',
  timestamps: true
});

export default Order;

File: flutter/cafeteria_app/lib/models/order.dart
class OrderRequest {
  final String orderType; // 'delivery', 'pickup', 'dine_in'
  final List<Map<String, dynamic>> items;
  final double totalAmount;
  final String? deliveryAddress;
  final String customerName;
  final String customerPhone;
  final String customerEmail;
  final String paymentMethod; // 'card', 'cash', 'sbp'
  final String? specialInstructions;

  OrderRequest({
    required this.orderType,
    required this.items,
    required this.totalAmount,
    this.deliveryAddress,
    required this.customerName,
    required this.customerPhone,
    required this.customerEmail,
    required this.paymentMethod,
    this.specialInstructions,
  });

  Map<String, dynamic> toJson() => {
        'order_type': orderType,
        'items': items,
        'total_amount': totalAmount,
        'delivery_address': deliveryAddress,
        'customer_name': customerName,
        'customer_phone': customerPhone,
        'customer_email': customerEmail,
        'payment_method': paymentMethod,
        'special_instructions': specialInstructions,
      };
}

class OrderResponse {
  final int id;
  final String orderNumber;
  final String status;

  OrderResponse({
    required this.id,
    required this.orderNumber,
    required this.status,
  });

  factory OrderResponse.fromJson(Map<String, dynamic> json) {
    return OrderResponse(
      id: json['id'] as int,
      orderNumber: json['order_number'] as String,
      status: json['status'] as String? ?? 'pending',
    );
  }
}
